### PR TITLE
functiaonl Test after whitelisted

### PR DESCRIPTION
**Pull request by hmcts-shajahan**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by hmcts-kodatala — APPROVED** [review]
OK

File: src/main/java/uk/gov/hmcts/reform/hmc/api/config/UserAuthTokenGenerator.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    @Value("${idam.refDataUserAuth.scope}")
+    private String scope;
+
+    RestTemplate restTemplateIdam = new RestTemplate();
+
+    @Value("${idam.refDataUserAuth.userIdamBaseurl}")
+    private String idamBaseUrl;
+
+    private static final String TOKEN_ENDPOINT_PATH = "/o/token";
```
Comment (hmcts-kodatala) [inline src/main/java/uk/gov/hmcts/reform/hmc/api/config/UserAuthTokenGenerator.java:47]: reuse the idam api services which already there

**Review by hmcts-kodatala — COMMENTED** [review]
find review comments and merge from master  to this pr before merge

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -171,13 +171,18 @@ withPipeline(type, product, component) {
     env.ENVIRONMENT = "preview"
     env.CASE_API_URL = "https://fis-hmc-api-pr-${CHANGE_ID}.service.core-compute-preview.internal"
     env.CASE_API_TEST_URL = "https://fis-hmc-api-pr-${CHANGE_ID}.service.core-compute-preview.internal"
+    env.CASE_DATA_STORE_BASEURL="http://ccd-data-store-api-demo.service.core-compute-demo.internal"
+    env.AUTH_PROVIDER_SERVICE_CLIENT_BASEURL= "http://rpe-service-auth-provider-demo.service.core-compute-demo.internal"
+    env.HMC_CFT_HEARING_BASEURL = 'http://hmc-cft-hearing-service-demo.service.core-compute-demo.internal/hearings/'
```
[the remark this answers is not in the document]
Comment (hmcts-shajahan) [reply Jenkinsfile_CNP:176]: done

File: src/main/java/uk/gov/hmcts/reform/hmc/api/services/RefDataServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                            userAuthTokenGenerator.getSecurityTokens(),
+                            authTokenGenerator.generate(),
+                            epimmsId);
+            LOG.info("RefData call completed successfully" + courtDetailList);
+            List<CourtDetail> filteredCourtDetail =
+                    courtDetailList.stream()
+                            .filter(e -> "18".equals(e.getCourtTypeId()))
+                            .collect(Collectors.toList());
+            if (!filteredCourtDetail.isEmpty()) {
+                courtDetail = filteredCourtDetail.get(0);
```
Comment (hmcts-kodatala) [inline src/main/java/uk/gov/hmcts/reform/hmc/api/services/RefDataServiceImpl.java:51]: only one record is fine if come more than one court details for court type??

File: src/main/java/uk/gov/hmcts/reform/hmc/api/services/RefDataServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+@Service
+@SuppressWarnings("unchecked")
+public class RefDataServiceImpl implements RefDataService {
+
+    @Autowired AuthTokenGenerator authTokenGenerator;
+    @Autowired UserAuthTokenGenerator userAuthTokenGenerator;
+    @Autowired RefDataApi refDataApi;
+    private static final Logger LOG = LoggerFactory.getLogger(RefDataServiceImpl.class);
+
```
Comment (hmcts-kodatala) [inline src/main/java/uk/gov/hmcts/reform/hmc/api/services/RefDataServiceImpl.java:27]: use @slf4j for loogining

File: src/main/java/uk/gov/hmcts/reform/hmc/api/config/ServiceBusConfiguration.java [hunk 5]
```diff
@@ -89,8 +92,17 @@ public CompletableFuture<Void> onMessageAsync(IMessage message) {
                         String messageReceived =
                                 mapper.writeValueAsString(
                                         mapper.readValue(body.get(0), Hearing.class));
-                        log.debug(messageReceived);
                         Hearing hearing = mapper.readValue(body.get(0), Hearing.class);
```
Comment (hmcts-kodatala) [inline src/main/java/uk/gov/hmcts/reform/hmc/api/config/ServiceBusConfiguration.java:95]: @Profile("!test"). why we use this here?

File: src/main/resources/application.yaml [hunk 6]
```diff
@@ -44,6 +44,11 @@ hearing:
   api:
     url: ${HEARING_API_URL:http://rd-commondata-api-demo.service.core-compute-demo.internal}
 
+
+ref_data:
+  baseUrl: ${REF_DATA_URL:http://rd-location-ref-api-demo.service.core-compute-demo.internal/}
+
```
Comment (hmcts-kodatala) [inline src/main/resources/application.yaml:50]: REF_DATA_URL need to set it in values.yml

File: src/main/resources/application.yaml [hunk 7]
```diff
@@ -53,13 +58,20 @@ idam:
   api:
     url: ${IDAM_API_BASEURL:https://idam-api.demo.platform.hmcts.net}
   serviceAccount:
+    clientId: ${HMC_FIS_HEARING_SERVICE_CLIENT_ID:hmc-fis-hearing-service}
+    clientSecret: ${HMC_FIS_HEARING_SERVICE_CLIENT_SECRET:dummy}
+    grantType: ${IDAM_HMC_COMPONENT_GRANT_TYPE:password}
```
Comment (hmcts-kodatala) [inline src/main/resources/application.yaml:63]: need to set it in values.yml  if these are new or not set in

**Review by hmcts-kodatala — CHANGES_REQUESTED** [review]
pls check n update

File: src/main/java/uk/gov/hmcts/reform/hmc/api/config/ServiceBusConfiguration.java (hunk 5 above)
Comment (hmcts-shajahan) [reply src/main/java/uk/gov/hmcts/reform/hmc/api/config/ServiceBusConfiguration.java:96]: so it wont be considered for bean creation

File: src/main/java/uk/gov/hmcts/reform/hmc/api/config/UserAuthTokenGenerator.java (hunk 1 above)
Comment (hmcts-shajahan) [reply src/main/java/uk/gov/hmcts/reform/hmc/api/config/UserAuthTokenGenerator.java]: resolved

File: src/main/java/uk/gov/hmcts/reform/hmc/api/services/RefDataServiceImpl.java (hunk 4 above)
Comment (hmcts-shajahan) [reply src/main/java/uk/gov/hmcts/reform/hmc/api/services/RefDataServiceImpl.java:28]: chnaged

File: src/main/java/uk/gov/hmcts/reform/hmc/api/services/RefDataServiceImpl.java (hunk 3 above)
Comment (hmcts-shajahan) [reply src/main/java/uk/gov/hmcts/reform/hmc/api/services/RefDataServiceImpl.java:52]: For that court_type =18 only one record will come

File: src/main/resources/application.yaml (hunk 6 above)
Comment (hmcts-shajahan) [reply src/main/resources/application.yaml:50]: done

File: src/main/resources/application.yaml (hunk 7 above)
Comment (hmcts-shajahan) [reply src/main/resources/application.yaml]: done

**Review by hmcts-kodatala — APPROVED** [review]
OK

**Pull request merged**